### ISSUE TITLE
`missing_errors_doc` warns on test functions when `check-private-items = true`

### ISSUE THREAD

**Issue by GrantGryczan** [rust-lang/rust-clippy#13391]
### Summary

When `check-private-items = true` is set in `clippy.toml`, `missing_errors_doc` warnings can appear on test functions, even though there's no reason to document the meanings of errors returned by test functions, since these errors won't used by other code.

### Lint Name

missing_errors_doc

### Reproducer

I tried this code:

```rust
#[warn(clippy::missing_errors_doc)]
#[test]
fn test() -> Result<(), ()> {
    Ok(())
}
```

I saw this happen:

```
warning: docs for function returning `Result` missing `# Errors` section
 --> src/main.rs:7:1
  |
7 | fn test() -> Result<(), ()> {
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#missing_errors_doc
note: the lint level is defined here
 --> src/main.rs:5:8
  |
5 | #[warn(clippy::missing_errors_doc)]
  |        ^^^^^^^^^^^^^^^^^^^^^^^^^^


```

I expected to see this happen:

No warning.

### Version

```text
rustc 1.81.0 (eeb90cda1 2024-09-04)
binary: rustc
commit-hash: eeb90cda1969383f56a2637cbd3037bdf598841c
commit-date: 2024-09-04
host: x86_64-pc-windows-msvc
release: 1.81.0
LLVM version: 18.1.7
```

**Comment by alex-semenyuk** [comment]
Don't see this issue at latest version

**Comment by GrantGryczan** [comment]
I'm confused, I'm still seeing this issue even in Clippy 1.82.

**Comment by TomFryersMidsummer** [comment]
I'm still seeing this in 1.91, too.

**Comment by TomFryersMidsummer** [comment]
`Cargo.toml`:
```toml
[package]
name = "clippy-bug"
version = "0.1.0"
edition = "2024"
```
`clippy.toml`:
```toml
check-private-items = true
```
`src/main.rs`:
```rust
fn main() {}

#[warn(clippy::missing_errors_doc)]
#[test]
fn test() -> Result<(), ()> {
    Ok(())
}
```

```
$ cargo clippy --all-targets
warning: docs for function returning `Result` missing `# Errors` section
 --> src/main.rs:5:1
  |
5 | fn test() -> Result<(), ()> {
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/rust-1.91.0/index.html#missing_errors_doc
note: the lint level is defined here
 --> src/main.rs:3:8
  |
3 | #[warn(clippy::missing_errors_doc)]
  |        ^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: `clippy-bug` (bin "clippy-bug" test) generated 1 warning
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.01s
```

**Comment by y21** [comment]
The test case that was added in #13610 doesn't actually do anything because there's no `//@compile-flags:--test` annotation, so it only doesn't produce a warning because the `#[test]` function is cfg'd out. Reopening in that case since it's indeed still an issue.
